tests/ut/plugin_demo/customIdx/customIdx.cc

```diff
@@ -0,0 +1,503 @@
+                std::memcpy(result.get() + qIdx * dim, base_vec, dim * sizeof(VecT));
+            }
+            return knowhere::GenResultDataSet(row, dim, result.release());
+        } catch (const std::exception& e) {
+            LOG_KNOWHERE_ERROR_ << "GetVectorByIds on CustomIdx error : " << e.what();
+            return knowhere::expected<knowhere::DataSetPtr>::Err(knowhere::Status::plugin_index_error, e.what());
+        }
+    }
+
+    bool
+    HasRawData(const std::string& metric_type) const override {
+        return true;
+    }
+
+    knowhere::expected<knowhere::DataSetPtr>
+    GetIndexMeta(const knowhere::Config& cfg) const override {
+        return knowhere::expected<knowhere::DataSetPtr>::Err(knowhere::Status::not_implemented,
+                                                             "GetIndexMeta not implemented");
+    }
+
+    knowhere::Status
+    Serialize(knowhere::BinarySet& binset) const override {
+        std::shared_ptr<uint8_t[]> binary(new uint8_t[Size()]);
+        int offset = 0;
+
+        std::memcpy(binary.get() + offset, &row_, sizeof(int64_t));
+        offset += sizeof(int64_t);
+
+        std::memcpy(binary.get() + offset, &dim_, sizeof(int64_t));
+        offset += sizeof(int64_t);
+
+        std::memcpy(binary.get() + offset, data_.get(), row_ * dim_ * sizeof(VecT));
+        offset += row_ * dim_ * sizeof(VecT);
+
+        binset.Append(Type(), binary, offset);
+        return knowhere::Status::success;
+    }
+
+    knowhere::Status
+    Deserialize(const knowhere::BinarySet& binset, const knowhere::Config& config) override {
+        auto binary = binset.GetByName(Type());
+        int offset = 0;
+
+        std::memcpy(&row_, binary->data.get() + offset, sizeof(int64_t));
+        offset += sizeof(int64_t);
+
+        std::memcpy(&dim_, binary->data.get() + offset, sizeof(int64_t));
+        offset += sizeof(int64_t);
+
+        if (!data_) {
+            data_ = std::make_unique<VecT[]>(row_ * dim_);
+        }
+        std::memcpy(data_.get(), binary->data.get() + offset, row_ * dim_ * sizeof(VecT));
+        offset += row_ * dim_ * sizeof(VecT);
+
+        return knowhere::Status::success;
+    }
+
+    knowhere::Status
+    DeserializeFromFile(const std::string& filename, const knowhere::Config& config) override {
+        return knowhere::Status::not_implemented;
+    }
+
+    std::unique_ptr<knowhere::BaseConfig>
+    CreateConfig() const override {
+        return std::make_unique<demo::CustomIdxConfig>();
+    }
+    int64_t
+    Dim() const override {
+        return dim_;
+    }
+    int64_t
+    Size() const override {
+        return dim_ * row_ * sizeof(VecT) + sizeof(int64_t) * 2;
+    }
+    int64_t
+    Count() const override {
+        return row_;
+    }
+    std::string
+    Type() const override {
+        return "CustomIdx";
+    }
+
+ private:
+    int64_t row_;
+    int64_t dim_;
+    std::unique_ptr<VecT[]> data_;
+};
+}  // namespace demo
+extern "C" {
```

Comment:
why is `extern "C"` used here?

Reply:
To avoid compiler modify the function name in c++ style.